sos/report/plugins/grafana.py

```diff
@@ -19,29 +19,53 @@ class Grafana(Plugin, IndependentPlugin):
 
     packages = ('grafana',)
 
+    def _is_snap_installed(self):
+        grafana_pkg = self.policy.package_manager.pkg_by_name('grafana')
+        if grafana_pkg:
+            return grafana_pkg['pkg_manager'] == 'snap'
+        return False
+
     def setup(self):
-        if self.get_option("all_logs"):
-            self.add_copy_spec("/var/log/grafana/*.log*")
+        self._is_snap = self._is_snap_installed()
+        if self._is_snap:
+            grafana_cli = "grafana.grafana-cli"
+            log_path = "/var/snap/grafana/common/data/log/"
+            config_path = "/var/snap/grafana/current/conf/grafana.ini"
+
         else:
-            self.add_copy_spec("/var/log/grafana/*.log")
+            grafana_cli = "grafana-cli"
+            log_path = "/var/log/grafana/"
+            config_path = "/etc/grafana/"
 
-        self.add_cmd_output([
-            "grafana-cli plugins ls",
-            "grafana-cli plugins list-remote",
-            "grafana-cli -v",
-            "grafana-server -v",
-        ])
+        add_cmds = [
+            f'{grafana_cli} plugins ls',
+            f'{grafana_cli} plugins list-remote',
+            'snap info grafana' if self._is_snap else None,
+            f'{grafana_cli} -v' if not self._is_snap else None,
+            'grafana-server -v' if not self._is_snap else None
+        ]
+        self.add_cmd_output(list(filter(None, add_cmds)))
+
+        log_file_pattern = "*.log*" if self.get_option("all_logs") else "*.log"
 
         self.add_copy_spec([
-            "/etc/grafana/",
-            "/etc/sysconfig/grafana-server",
+            log_path + log_file_pattern,
+            config_path,
+            "/etc/sysconfig/grafana-server" if not self._is_snap else None
```

Comment:
`add_copy_spec()` also handles non-existing paths gracefully, so we can just have it listed here and it will be ignored for snap installations.